Sources/LeafKit/LeafLexer.swift

```diff
@@ -79,176 +109,243 @@ struct TemplateSource {
+            default: break
+        }
+        
+        // Complex Parameter lexing situations - enhanced to allow non-whitespace separated values
+        // Complicated by overlap in acceptable isValidInParameter characters between possible types
+        // Process from most restrictive options to least to help prevent overly aggressive tokens
+        // Possible results, most restrictive to least
+        // * Operator
+        // * Constant(Int)
+        // * Constant(Double)
+        // * Keyword
+        // * Tag
+        // * Variable
+        
+        // if current character isn't valid for any kind of parameter, something's majorly wrong
+        guard current.isValidInParameter else {
+            throw LexerError(.invalidParameterToken(current), src: src, lexed: lexed)
+        }
+
+        // Test for Operator first - this will only handle max two character operators, not ideal
+        // Can't switch on this, MUST happen before trying to read tags
+        if current.isValidOperator {
+            var op = Operator(rawValue: String(current) + String(src.peek()!))
+            if op == nil { op = Operator(rawValue: String(current)) } else { src.pop() }
+            if op != nil { return .parameter(.operator(op!)) }
+        }
+        
+        // Test for numerics next. This is not very intelligent but will read base2/8/10/16
+        // for Ints and base 10/16 for decimal through native Swift initialization
+        // Will not adequately decay to handle things like `0b0A` and recognize as invalid.
+        if current.canStartNumeric {
+            var testInt: Int?
+            var testDouble: Double?
+            var radix: Int?
+            var sign = 1
+            
+            let next = src.peek()!
+            let peekRaw = String(current) + (src.peekWhile { $0.isValidInNumeric } ?? "")
+            var peekNum = peekRaw.replacingOccurrences(of: String(.underscore), with: "")
+            // We must be immediately preceeded by a minus to flip the sign
+            // And only flip back if immediately preceeded by a const, tag or variable
+            // (which we assume will provide a numeric). Grammatical errors in the
+            // template (eg, keyword-numeric) may throw here
+            if case .parameter(let p) = lexed[offset - 1], case .operator(let op) = p, op == .minus {
+                switch lexed[offset - 2] {
+                    case .parameter(let p):
+                        switch p {
+                            case .constant,
+                                 .tag,
+                                 .variable: sign = 1
+                            default: throw LexerError(.invalidParameterToken("-"), src: src)
+                        }
+                    case .stringLiteral: throw LexerError(.invalidParameterToken("-"), src: src)
+                    default: sign = -1
+                }
+            }
+            
+            switch (peekNum.contains(.period), next, peekNum.count > 2) {
```

Comment:
As far as I can tell, there's no where that actually defaults `radix` to `10` anywhere. Either make `nil` imply that and remove the `radix != 10` check below as it will never be `10` anyway, or make it explicitly 10 by default and make it non-optional.

Reply:
Good call, radix initialized to nil and optional test for != 10 dumped since this block is explicitly supposed to be only for `0b/0o/0x` notation